skin/frontend/base/default/js/sd_recentlyviewed/recent.js

```diff
@@ -62,13 +62,23 @@ SomethingDigitalRecentlyViewed.prototype = {
 		var items = this.getLoadedStorage();
 
 		return items.map(function(a){
+			if($(document.body).hasClassName('catalog-product-view')){
+
+				var currentProductId = $$('#product_addtocart_form input[name=product]')[0].value;
```

Comment:
This likely should be configurable (like the other html selectors) from with in the admin to maintain theme compatibility.